Loop/Extensions/DeviceDataManager+BolusEntryViewModelDelegate.swift

```diff
@@ -59,8 +59,8 @@ extension DeviceDataManager: BolusEntryViewModelDelegate {
         return pumpManager != nil
     }
     
-    var preferredGlucoseUnit: HKUnit? {
-        return glucoseStore.preferredUnit
+    var preferredGlucoseUnit: HKUnit {
+        return glucoseStore.preferredUnit!
```

Comment:
Yeah, this one I think should be reverted; the real fix is that we shouldn't even bring up either bolus interface if we don't have units; we need to know units to do these interfaces.

Reply:
Actually, ended up defaulting to mg/dL at the delegate layer; that's what these interfaces do. Missing units here is due to some storage issue or unavailability.

Reply:
Not a fan of this but I'm ready to move on.

Reply:
We should actively work on pushing optionality to the earliest point possible, as it simplifies reasoning about later code. Each time we propagate an optional to a new function/type, it doubles the state space that needs to be considered where those are used.

Reply:
I agree, but your statement doesn't mean that force-unwrapping is the best way to deal with optionality.

Reply:
Did you see that I wasn't force unwrapping this? I assumed you did, and your "Not a fan" comment was about that, but based on your latest comment, I think maybe you didn't see the change to use mg/dL if the setting was not available? The UIs were doing this in various places, so now it is unified at a lower level at the delegate.

Reply:
I'm super confused.  Github is showing you force-unwrapping here.  That's what I was not a fan of. What am I missing?

Reply:
Hmm, here's a link to the latest change in the PR: https://github.com/tidepool-org/Loop/pull/280/files#diff-164ef8871fe9a7559912cd4386693e7204df9ce47605e3ef3ea21f2a49bddc83R63, and here's a link to it in the branch: https://github.com/tidepool-org/Loop/blob/ps2/LOOP-1869/simple-bolus-calculator/Loop/Extensions/DeviceDataManager%2BBolusEntryViewModelDelegate.swift#L63

Reply:
Crazy!  Okay, well, then never mind then.  It is very confusing navigating github then, because for me it is showing 
```swift
    var preferredGlucoseUnit: HKUnit {
        return glucoseStore.preferredUnit!
```